text_collector_examples/nvme_metrics.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+set -eu
+
+# Dependencies: nvme-cli (package)
+# Based on code from
+# - https://github.com/prometheus/node_exporter/blob/master/text_collector_examples/smartmon.sh
+# - https://github.com/vorlon/check_nvme/blob/master/check_nvme.sh
+# Author: Henk <henk@wearespindle.com>
+
+output_format_awk="$(
+  cat <<'OUTPUTAWK'
+BEGIN { v = "" }
+v != $1 {
+  print "# HELP nvme_" $1 " SMART metric " $1;
+  print "# TYPE nvme_" $1 " gauge";
+  v = $1
+}
+{print "nvme_" $0}
+OUTPUTAWK
+)"
+
+format_output() {
+  sort | awk -F'{' "${output_format_awk}"
+}
+
+# Get the nvme-cli version
+nvme_version="$(nvme version | awk '$1 == "nvme" {print $3}')"
+echo "nvmecli{version=\"${nvme_version}\"} 1" | format_output
+
+# Cut device name to /dev/nvmeX
+device_list="$(nvme list | awk '/^\/dev/{print $1}' | cut -c1-10)"
+
+for device in ${device_list}; do
+  CHECK=$(nvme smart-log "${device}")
```

Comment:
Minor bash style nit-pic. Let's use a lowercase variable here. Also double-quote the `$()`

Reply:
Fixed.